### PR TITLE
Add method to retrieve the namespace names for Cosmos Admin

### PR DESCRIPTION
**Pull request by Torch3333**
This implement the `admin.getNamespaceNames()` for the CosmosAdmin .

### CODE REVIEW

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Review by brfrn169 — COMMENTED** [review]
LGTM overall. Left a question. Please take a look when you have time!

I also have the following questions:

- Should we have a method to repair the namespaces metadata table?
- Just a confirmation, for backward compatibility, you are going to add something like upgrade command in the Schema Loader to load the existing namespaces to the namespaces metadata table, right?

Please take a look!

File: core/src/main/java/com/scalar/db/storage/cosmos/CosmosAdmin.java [hunk 1]
```diff
@@ -319,12 +320,24 @@ private void deleteTableMetadata(String namespace, String table) throws Executio
...
+      // namespaces
+      if (!namespacesContainer
+          .queryItems(
+              "SELECT 1 FROM container OFFSET 0 LIMIT 1",
+              new CosmosQueryRequestOptions(),
+              Object.class)
+          .stream()
+          .findFirst()
+          .isPresent()) {
+        client.getDatabase(metadataDatabase).delete();
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/storage/cosmos/CosmosAdmin.java:339]: Don't we need to delete the namespaces container before deleting the metadata database?

File: core/src/main/java/com/scalar/db/storage/cosmos/CosmosAdmin.java [hunk 2]
```diff
@@ -53,7 +53,8 @@ public class CosmosAdmin implements DistributedStorageAdmin {
   public static final String DEFAULT_NO_SCALING = "false";
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/storage/cosmos/CosmosAdmin.java:53]: Don't we need to change the implementation of `namespaceExists()` as well?

Comment (brfrn169) [reply core/src/main/java/com/scalar/db/storage/cosmos/CosmosAdmin.java:53]: Ah, we may need to change `namespaceExists()` in `JdbcAdmin` as well.

File: core/src/main/java/com/scalar/db/storage/cosmos/CosmosAdmin.java [hunk 3]
```diff
@@ -319,12 +320,24 @@ private void deleteTableMetadata(String namespace, String table) throws Executio
 
   @Override
   public void dropNamespace(String namespace) throws ExecutionException {
-    if (!databaseExists(namespace)) {
-      throw new ExecutionException("the database does not exist");
-    }
```
[the remark this answers is not in the document]
Comment (Torch3333) [reply core/src/main/java/com/scalar/db/storage/cosmos/CosmosAdmin.java:324]: I feel like this check was not necessary as the deletion call `client.getDatabase(namespace).delete();` will throw an exception if the database does not exists.

File: core/src/main/java/com/scalar/db/storage/cosmos/CosmosAdmin.java (hunk 1 above)
Comment (Torch3333) [reply core/src/main/java/com/scalar/db/storage/cosmos/CosmosAdmin.java:339]: Deleting the database will delete all its container so deleting the container first is not required.

File: core/src/main/java/com/scalar/db/storage/cosmos/CosmosAdmin.java (hunk 2 above)
Comment (Torch3333) [reply core/src/main/java/com/scalar/db/storage/cosmos/CosmosAdmin.java:53]: Yes, good point thank you. I will update `namespaceExists()` in all the admins.

**Comment by Torch3333** [comment]
@brfrn169 
>Should we have a method to repair the namespaces metadata table?

I think so. Concretely, this method would be an alias for `admin.createNamespace("my_ns", ifNotExists=true, options)`. 
We could then update the Schema Loader `--repair-all` command to also execute `admin.repairNamespace()`   

> * Just a confirmation, for backward compatibility, you are going to add something like upgrade command in the Schema Loader to load the existing namespaces to the namespaces metadata table, right?

Yes. I started working on implementing the `admin.upgrade()` method for now.

File: core/src/main/java/com/scalar/db/storage/cosmos/CosmosAdmin.java (hunk 2 above)
Comment (Torch3333) [reply core/src/main/java/com/scalar/db/storage/cosmos/CosmosAdmin.java:53]: I revised the `namespaceExists()` implementation in the latest commit. Please have a look when you can.

**Comment by Torch3333** [comment]
I changed the PR target branch to the [feat/namespaces_management](https://github.com/scalar-labs/scalardb/tree/feat/namespaces_management) branch that will contain all the namespaces management-related changes.

Sorry for the trouble. Please have a look again

**Review by brfrn169 — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**